config/lib/mongoose.js

```diff
@@ -24,25 +25,28 @@ module.exports.connect = function (cb) {
 
   mongoose.Promise = config.db.promise;
 
-  var db = mongoose.connect(config.db.uri, config.db.options, function (err) {
-    // Log Error
-    if (err) {
-      console.error(chalk.red('Could not connect to MongoDB!'));
-      console.log(err);
-    } else {
+  var options = _.merge(config.db.options || {}, { useMongoClient: true });
 
+  mongoose
+    .connect(config.db.uri, options)
+    .then(function (connection) {
       // Enabling mongoose debug mode if required
       mongoose.set('debug', config.db.debug);
 
       // Call callback FN
-      if (cb) cb(db);
-    }
-  });
+      if (cb) cb(connection.db);
+    })
+    .catch((err) => {
```

Comment:
Everything you just said seems very reasonable to me. I'd like to discuss the introduction of ES6 as well. I think it started becoming a habit for me, since I'm using some features like array functions in my projects.

I'll remove my ES6 usage from this PR.